src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EmbeddedDataContainerNode.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Internal.TypeSystem;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    /// <summary>
+    /// Represents a node that contains a set of embedded objects. The main function is 
+    /// to serve as a base class, providing symbol name boundaries and node ordering.
+    /// </summary>
+    public abstract class EmbeddedDataContainerNode : ObjectNode
+    {
+        private ObjectAndOffsetSymbolNode _startSymbol;
+        private ObjectAndOffsetSymbolNode _endSymbol;
+
+        public ObjectAndOffsetSymbolNode StartSymbol => _startSymbol;
+        public ObjectAndOffsetSymbolNode EndSymbol => _endSymbol;
+
+        protected EmbeddedDataContainerNode(string startSymbolMangledName, string endSymbolMangledName)
+        {
+            _startSymbol = new ObjectAndOffsetSymbolNode(this, 0, startSymbolMangledName, true);
+            _endSymbol = new ObjectAndOffsetSymbolNode(this, 0, endSymbolMangledName, true);
+        }
+
+        protected override int ClassCode => (int)ObjectNodeOrder.EmbeddedDataContainerNode;
+
+        protected override int CompareToImpl(ObjectNode other, TypeSystemComparer comparer, NameMangler nameMangler)
+        {
+            return StartSymbol.GetMangledName(nameMangler).CompareTo(((EmbeddedDataContainerNode)other).StartSymbol.GetMangledName(nameMangler));
```

Comment:
I don't have access to an actual development environment (typing this from a tablet), but I would at least consider ISortableSymbolNode or something. Once we start sorting on mangled names, we might as well sort *everything* based on the mangled name and strip this change to basically just a new class that does that, while caching the name on the side...